### DISCUSSION TITLE
Set LSP server to other than default

### DISCUSSION THREAD

**Discussion by a-berg** [LunarVim/LunarVim#2350] [Q&A] [answered]
Hello, I'm kinda new to neovim. I am trying to set the (**python**) LSP server to jedi (instead of the default pyright) but I don't know how to do it. I have searched for both base neovim and in the lunarvim docs but it is unclear to me... 😕

**Comment by abzcoding** [comment, marked as answer]
add this to your `~/.config/lvim/config.lua`
```lua
vim.list_extend(lvim.lsp.override, {  "pyright" })
```

and then create a file called `~/.config/lvim/ftplugin/python.lua`
```lua
local opts = {
  root_dir = function(fname)
    local util = require "lspconfig.util"
    local root_files = {
      "pyproject.toml",
      "setup.py",
      "setup.cfg",
      "requirements.txt",
      "Pipfile",
      "manage.py",
      "pyrightconfig.json",
    }
    return util.root_pattern(unpack(root_files))(fname) or util.root_pattern ".git"(fname) or util.path.dirname(fname)
  end,
  single_file_support = true,
}

local servers = require "nvim-lsp-installer.servers"
local server_available, requested_server = servers.get_server "jedi_language_server"
if server_available then
  opts.cmd_env = requested_server:get_default_options().cmd_env
end

require("lvim.lsp.manager").setup("jedi_language_server", opts)
```

**Reply by a-berg** [reply]
Follow-up, though: why that folder structure in particular? I have seen other neovim users do stuff like `nvim/lua/plugins.lua`, or multiple files (which I like, in fact, I was looking to break up my `lvim` config into multiple parts)

**Reply by abzcoding** [reply]
oh because `ftplugin` folder is a special folder
<img width="647" alt="Screen Shot 2022-03-15 at 9 28 26 PM" src="https://user-images.githubusercontent.com/10992695/158441865-8d297764-e608-4a6f-92cd-3909366b5f9f.png">

and files inside it will load based on filetype so
`ftplugin/python.lua` will only load when you open a `python` file and
`ftplugin/c.lua` will only load when you open a `c` file

if you want to split your config into a more structure one i would suggest checking out [my config](https://github.com/abzcoding/lvim)
